proxy_collection.go

```diff
@@ -61,6 +66,49 @@ func (collection *ProxyCollection) AddOrReplace(proxy *Proxy, start bool) error
 	return nil
 }
 
+func (collection *ProxyCollection) PopulateJson(data io.Reader) ([]*Proxy, error) {
```

Comment:
This mimics the `AddToxicJson` and `UpdateToxicJson` functions in `toxic_collection.go` fairly well, where there's some go+json hackery going on to get things in the right format.
The `input` struct is fairly json specific since getting default vaults in a json array is a bit tricky. I'd rather not be passing it around.
At least this way it's isolated fairly well to one function.